### ISSUE TITLE
Write text macro types wrong characters

### ISSUE THREAD

**Issue by sullrich84** [UltimateHackingKeyboard/agent#1962]
I've created a macro that should type my email address (e.g. `timmy@example.com`). However the resulting output is `timmz"example.com`. The issue occurs on Mac OSX using a german keyboard layout. 

It seems that the original characters get mapped to ANSI keycode. A possible work around is to separately capture every keystroke. Is it possible to change the character mapping somehow?

**Comment by mondalaci** [comment]
I'm sorry to say that [the core of the issue](https://ultimatehackingkeyboard.com/blog/2018/06/23/how-can-i-type-accented-characters-with-my-uhk) is so problematic that we don't plan to implement a better solution.

**Comment by decibyte** [comment]
I came here to report the same, but I do respect the decision to not fix it.

Let me instead provide the workaround I came up with: Split the macro into 3 steps.

1) "Write text" everything before the "@".
2) "Tap key" the key combination that will get you an "@" symbol (e.g. in my case, with a Danish layout, it's `RAlt(Gr) + 2 @`).
3) "Write text" everything after the "@".

A couple of screenshots of what it looks like in Agent:

![image](https://github.com/user-attachments/assets/3561f3e1-b7a8-4eef-b9e1-f8b03686bafa)

![image](https://github.com/user-attachments/assets/4c59d5d0-fd45-40b2-a9c6-17631919f4c3)
